src/Metal/MTLDevice.cs

```diff
@@ -42,6 +60,28 @@ public static partial class MTLDevice {
 				return system_default;
 			}
 		}
+
+#if MONOMAC
+		[Mac (10,13, onlyOn64: true), NoiOS, NoWatch, NoTV]
+		[DllImport (Constants.MetalLibrary)]
+		static extern IMTLDevice[] MTLCopyAllDevicesWithObserver (IntPtr observer, MTLDeviceNotificationHandler handler);
```

Comment:
So problem here.  To marshal the array from an IntPtr you need to know the length of the array (unless there's something I don't know about).  This method doesn't include said length anywhere I see, so how would I do that?